### PR TITLE
CEXT-934 - Add docs for event field data minimization

### PR DESCRIPTION
**Pull request by keharper**
## Description

Adds new sections to Module Development section and updates the `events:subscribe` command. 

Staging build: https://adobedocs.github.io/commerce-events/events/module-development/

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/events/commands.md [hunk 1]
```diff
@@ -84,24 +84,30 @@ where:
...
 
+### Options
+
+`--fields <field_name>` Required. An event field to transmit to the Adobe App Builder application. You can specify this option multiple times. Each instance can contain only one field name.
+
 ### Example
 
 ```bash
-bin/magento events:subscribe observer.customer_login
+bin/magento events:subscribe com.adobe.commerce.observer.catalog_product_save_after --fields sku --fields stock_data.qty
```
Comment (oshmyheliuk) [inline src/pages/events/commands.md:104]: `com.adobe.commerce` prefix is added automatically and must not be presented in the `event:subscribe` command. The command will return an error as it expects that event_code will start with **plugin** or **observer**.

File: src/pages/events/module-development.md [hunk 2]
```diff
@@ -14,30 +14,192 @@ You can programmatically register events using the following methods:
 *  Create an `io_events.xml` file in your module
 *  Declare them in the system `config.php` file
 
+For each event you register, you must define which fields to transmit to your App Builder application. The payload of an event can be massive. In addition, some events include sensitive or PCI compliance data by default. The following payload is returned for the `com.adobe.commerce.observer.catalog_product_save_after` event:
+
+```json
+{
+  position: 'camel:3f25472c-5f70-4f36-84be-2248d2e2d421.camel:9cd9055e-f56d-4fd3-8017-9db5ac1c2b3b.e501d499-cfc8-450d-bc42-8637bbf12601.0.1663803959.o3fdxogbstwwnxbxq2r-',
```
Comment (oshmyheliuk) [inline src/pages/events/module-development.md:21]: I would recommend leaving only the event value payload and removing everything else:
```
{
  value: {
    _edit_mode: true,
    store_id: '0',
    entity_id: '3',
    attribute_set_id: '4',
    type_id: 'simple',
    sku: 'test2',
    has_options: false,
    required_options: false,
    created_at: '2022-07-28 14:13:40',
    updated_at: '2022-09-06 20:37:25',
    row_id: '3',
    created_in: '1',
    updated_in: '2147483647',
    name: 'test2',
    meta_title: 'test2',
    meta_description: 'test2 ',
    page_layout: 'product-full-width',
    options_container: 'container2',
    url_key: 'test2',
    msrp_display_actual_price_type: '0',
    gift_message_available: '2',
    gift_wrapping_available: '2',
    is_returnable: '2',
    status: '1',
    visibility: '4',
    tax_class_id: '2',
    price: '123.000000',
    meta_keyword: 'test2',
    options: [],
    media_gallery: { images: [], values: [] },
    tier_price: [],
    tier_price_changed: 0,
    quantity_and_stock_status: {
      is_in_stock: '1',
      qty: '333'
    },
    category_ids: [ '4' ],
    is_salable: 1,
    stock_data: {
      item_id: '3',
      product_id: '3',
      stock_id: '1',
      qty: '333',
      min_qty: '0',
      use_config_min_qty: '1',
      is_qty_decimal: '0',
      backorders: '0',
      use_config_backorders: '1',
      min_sale_qty: '1',
      use_config_min_sale_qty: '1',
      max_sale_qty: '10000',
      use_config_max_sale_qty: '1',
      is_in_stock: '1',
      notify_stock_qty: '1',
      use_config_notify_stock_qty: '1',
      manage_stock: '1',
      use_config_manage_stock: '1',
      stock_status_changed_auto: '0',
      use_config_qty_increments: '1',
      qty_increments: '1',
      use_config_enable_qty_inc: '1',
      enable_qty_increments: '0',
      is_decimal_divided: 0,
      website_id: '0',
      deferred_stock_update: '0',
      use_config_deferred_stock_update: '1',
      type_id: 'simple',
      min_qty_allowed_in_shopping_cart: [Array]
    },
    use_config_gift_message_available: '1',
    use_config_gift_wrapping_available: '1',
    current_product_id: '3',
    affect_product_custom_options: '1',
    current_store_id: '0',
    product_has_weight: '1',
    is_new: '0',
    website_ids: { '1': '1' },
    url_key_create_redirect: 'test2',
    ignore_links_flag: false,
    can_save_custom_options: true,
    save_rewrites_history: true,
    can_save_bundle_selections: false,
    is_custom_option_changed: true,
    parent_id: 0,
    special_from_date_is_formated: true,
    custom_design_from_is_formated: true,
    news_from_date_is_formated: true,
    news_to_date_is_formated: true,
    is_changed_categories: false,
    is_changed_websites: false,
    force_reindex_eav_required: true
  }
}
```

File: src/pages/events/module-development.md [hunk 3]
```diff
@@ -14,30 +14,192 @@ You can programmatically register events using the following methods:
...
+    is_changed_categories: false,
+    is_changed_websites: false,
+    force_reindex_eav_required: true
+  },
+  source: 'demo.demo',
+  pipelineMessageId: '1663803954935:545088:VA7_A1:128:0'
+}
+```
+
+You define an array of fields to transmit in your configuration file. Specify any field within an event's `value` object to ensure it is transmitted to an application. If the field is part of a secondary object, such `stock_data` in the above example, use the format `<object-name>.field`. For example: `stock_data.product_id`.
```
Comment (jhadobe) [inline src/pages/events/module-development.md:139]:
```suggestion
You define an array of fields to transmit in your configuration file. Specify any field within an event's `value` object to ensure it is transmitted to an application. If the field is part of a secondary object, such as `stock_data` in the above example, use the format `<object-name>.field`. For example: `stock_data.product_id`.
```

File: src/pages/events/module-development.md [hunk 4]
```diff
@@ -14,30 +14,192 @@ You can programmatically register events using the following methods:
...
+    </event>
+</config>
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module-commerce-events-client/etc/io_events.xsd">
+
 </config>
 ```
 
-Run the [events:generate:module command](./commands.md#generate-a-commerce-module-based-on-a-list-of-subscribed-events) to generate the required plugins.
+The contents of an `commerce.observer.catalog_product_save_after` event are similar to the following:
```
Comment (jhadobe) [inline src/pages/events/module-development.md:185]:
```suggestion
The contents of a `commerce.observer.catalog_product_save_after` event are similar to the following:
```

**Review by jhadobe — APPROVED** [review]

File: src/pages/events/module-development.md [hunk 5]
```diff
@@ -14,50 +14,243 @@ You can programmatically register events using the following methods:
...
+        </fields>
+    </event>
+    <event name="plugin.magento.gift_wrapping.model.resource_model.wrapping.save">
+        <fields>
+            <field name="id" />
+            <field name="name" />
+            <field name="status" />
+        </fields>
+    </event>
+    <event name="plugin.magento.sales.api.invoice_item_repository.save">
```
Comment (oshmyheliuk) [inline src/pages/events/module-development.md:147]: Real fields for this event code, there are more but I think it should be enough for the example.
```
    <event name="plugin.magento.sales.api.invoice_item_repository.save">         
        <fields>
            <field name="entity_id" />
            <field name="parent_id" />
            <field name="base_price" />
            <field name="tax_amount" />
            <field name="base_row_total" />
            <field name="discount_amount" />
            <field name="qty" />
        </fields>
    </event>
```

**Review by oshmyheliuk — APPROVED** [review]

**Pull request merged**